tests/suite/snippets_filter_test.go

```diff
@@ -0,0 +1,337 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("SnippetsFilter", Ordered, Label("functional", "snippets-filter"), func() {
+	var (
+		files = []string{
+			"snippets-filter/cafe.yaml",
+			"snippets-filter/gateway.yaml",
+			"snippets-filter/grpc-backend.yaml",
+			"snippets-filter/cafe-routes.yaml",
+			"snippets-filter/grpc-route.yaml",
+		}
+
+		namespace = "snippets-filter"
+	)
+
+	BeforeAll(func() {
+		ns := &core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+
+		Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+		Expect(resourceManager.WaitForAppsToBeReady(namespace)).To(Succeed())
+	})
+
+	AfterAll(func() {
+		Expect(resourceManager.DeleteNamespace(namespace)).To(Succeed())
+	})
+
+	When("SnippetsFilters are applied to the resources", func() {
+		snippetsFilter := []string{
+			"snippets-filter/valid-sf.yaml",
+		}
+
+		BeforeAll(func() {
+			Expect(resourceManager.ApplyFromFiles(snippetsFilter, namespace)).To(Succeed())
+		})
+
+		AfterAll(func() {
+			Expect(resourceManager.DeleteFromFiles(snippetsFilter, namespace)).To(Succeed())
+		})
+
+		Specify("snippetsFilters are accepted", func() {
+			snippetsFilterNames := []string{
+				"all-contexts",
+				"grpc-all-contexts",
+			}
+
+			for _, name := range snippetsFilterNames {
+				nsname := types.NamespacedName{Name: name, Namespace: namespace}
+
+				err := waitForSnippetsFilterToBeAccepted(
+					nsname,
+					metav1.ConditionTrue,
+					ngfAPI.SnippetsFilterConditionReasonAccepted,
+				)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%s was not accepted", name))
+			}
+		})
+
+		Context("verify working traffic", func() {
+			It("should return a 200 response for HTTPRoute", func() {
+				port := 80
+				if portFwdPort != 0 {
+					port = portFwdPort
+				}
+				baseURL := fmt.Sprintf("http://cafe.example.com:%d%s", port, "/coffee")
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+			snippetsFilterFilePrefix := "/etc/nginx/includes/SnippetsFilter_"
+
+			mainContext := fmt.Sprintf("%ssmain_", snippetsFilterFilePrefix)
```

Comment:
Is that an extra s? Or am I mistaken?

```suggestion
			mainContext := fmt.Sprintf("%smain_", snippetsFilterFilePrefix)
```

Reply:
no the `%s` gets replaced by `snippetsFilterFilePrefix` .  

For more info check this out https://pkg.go.dev/fmt#pkg-variables

Reply:
yea, though currently you have `%ssmain_`, so it would get replaced with `snippetsFilterFilePrefix` + `smain_`, is that correct?

Reply:
omg you are so right. I am sorry missed it